File: kotlin/Client/src/main/kotlin/com/github/csandiego/pocopenapitools/client/openapitools/OpenApiToolsBookApi.kt
package com.github.csandiego.pocopenapitools.client.openapitools

import com.github.csandiego.pocopenapitools.openapitools.models.Book

interface OpenApiToolsBookApi {

    fun list(): Array<Book>

    fun create(book: Book)

    fun get(id: Int): Book

    fun update(id: Int, book: Book)

    fun delete(id: Int)
    
}

File: swift/Client/Sources/Lib/OpenApiToolsBookAPI.swift
import OpenApiTools

public protocol OpenApiToolsBookAPI {

    static func callGet(id: Int, completion: @escaping ((_ data: OpenApiTools.Book?, _ error: Error?) -> Void))

    static func create(book: OpenApiTools.Book, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))

    static func delete(id: Int, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))

    static func list(completion: @escaping ((_ data: [OpenApiTools.Book]?, _ error: Error?) -> Void))

    static func update(id: Int, book: OpenApiTools.Book, completion: @escaping ((_ data: Void?, _ error: Error?) -> Void))

}
